framework/i18n/MessageFormatter.php

```diff
@@ -274,19 +275,20 @@ protected function fallbackFormat($pattern, $args, $locale)
      */
     private static function tokenizePattern($pattern)
     {
+        $charset = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
Although, since by default `mb_strpos()` uses `mb_internal_encoding()`. Perhaps it is fine.